The terraform-provider-ibm_v1.27.1 plugin crashed

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform CLI and Terraform IBM Provider Version

```
$ terraform -v
Terraform v0.15.3
on darwin_amd64
+ provider registry.terraform.io/hashicorp/helm v2.2.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.3.2
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/hashicorp/time v0.7.2
+ provider registry.terraform.io/ibm-cloud/ibm v1.27.1
```

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* ibm_resource_group

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
data "ibm_resource_group" "group" {
  is_default = "true"
}
```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

### Panic Output

```
Stack trace from the terraform-provider-ibm_v1.27.1 plugin:

panic: runtime error: index out of range [0] with length 0

goroutine 56 [running]:
github.com/IBM-Cloud/terraform-provider-ibm/ibm.dataSourceIBMResourceGroupRead(0xc000124b00, 0x3481440, 0xc000762000, 0x47357f0, 0xc000500000)
	github.com/IBM-Cloud/terraform-provider-ibm/ibm/data_source_ibm_resource_group.go:67 +0x62f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc00093a700, 0x3973288, 0xc00011c740, 0xc000124b00, 0x3481440, 0xc000762000, 0x0, 0x0, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.1/helper/schema/resource.go:335 +0x1ee
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0xc00093a700, 0x3973288, 0xc00011c740, 0xc000d281a0, 0x3481440, 0xc000762000, 0xc000762000, 0xc000d281a0, 0x0, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.1/helper/schema/resource.go:558 +0xfd
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0xc00019d710, 0x3973288, 0xc00011c740, 0xc000d28100, 0xc00011c740, 0x100b705, 0x3309180)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.1/helper/schema/grpc_provider.go:1105 +0x4d6
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ReadDataSource(0xc00102ea00, 0x3973330, 0xc00011c740, 0xc000ddc140, 0xc00102ea00, 0xc001034330, 0xc000ea2ba0)
	github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:247 +0xe5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler(0x33b10c0, 0xc00102ea00, 0x3973330, 0xc001034330, 0xc000ad4120, 0x0, 0x3973330, 0xc001034330, 0xc000487140, 0x34)
	github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:416 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0005ece00, 0x398c9b8, 0xc0004d6480, 0xc000f22200, 0xc000a2fc20, 0x46d7cf0, 0x0, 0x0, 0x0)
	google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc0005ece00, 0x398c9b8, 0xc0004d6480, 0xc000f22200, 0x0)
	google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc00059f510, 0xc0005ece00, 0x398c9b8, 0xc0004d6480, 0xc000f22200)
	google.golang.org/grpc@v1.32.0/server.go:859 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd

Error: The terraform-provider-ibm_v1.27.1 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behavior

No errors

### Actual Behavior

The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadDataSource call.

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #0000

--- a/ibm/data_source_ibm_resource_group.go
+++ b/ibm/data_source_ibm_resource_group.go
@@ -17,16 +17,16 @@ func dataSourceIBMResourceGroup() *schema.Resource {
 
 		Schema: map[string]*schema.Schema{
 			"name": {
-				Description:   "Resource group name",
-				Type:          schema.TypeString,
-				Optional:      true,
-				ConflictsWith: []string{"is_default"},
+				Description:  "Resource group name",
+				Type:         schema.TypeString,
+				Optional:     true,
+				ExactlyOneOf: []string{"is_default", "name"},
 			},
 			"is_default": {
-				Description:   "Default Resource group",
-				Type:          schema.TypeBool,
-				Optional:      true,
-				ConflictsWith: []string{"name"},
+				Description:  "Default Resource group",
+				Type:         schema.TypeBool,
+				Optional:     true,
+				ExactlyOneOf: []string{"is_default", "name"},
 			},
 		},
 	}
@@ -62,7 +62,10 @@ func dataSourceIBMResourceGroupRead(d *schema.ResourceData, meta interface{}) er
 		grp, err = rsGroup.List(&resourceGroupQuery)
 
 		if err != nil {
-			return fmt.Errorf("Error retrieving default resource group: %s", err)
+			return fmt.Errorf("[ERROR] Error retrieving default resource group: %s", err)
+		}
+		if len(grp) < 1 {
+			return fmt.Errorf("[ERROR] No Default resource group found: %s", err)
 		}
 		d.SetId(grp[0].ID)
 
@@ -72,12 +75,10 @@ func dataSourceIBMResourceGroupRead(d *schema.ResourceData, meta interface{}) er
 		}
 		grp, err := rsGroup.FindByName(resourceGroupQuery, name)
 		if err != nil {
-			return fmt.Errorf("Error retrieving resource group %s: %s", name, err)
+			return fmt.Errorf("[ERROR] Error retrieving resource group %s: %s", name, err)
 		}
 		d.SetId(grp[0].ID)
 
-	} else {
-		return fmt.Errorf("Missing required properties. Need a resource group name, or the is_default true")
 	}
 
 	return nil

